lib/truncato/truncated_sax_document.rb

```diff
@@ -56,49 +96,65 @@ def end_document
 
   private
 
+  def remaining_length
+    max_length - estimated_length_with_tail
+  end
+
+  def estimated_length_with_tail
+    # byebug
+    @estimated_length + (@count_tail && @something_has_been_truncated ? tail_length : 0)
```

Comment:
parens around `@count_tail && @ something_has_been_truncated`? found it confusing upon first reading..also remove byebug comment above